platform/locale_linux.cpp

```diff
@@ -1,10 +1,48 @@
 #include "platform/locale.hpp"
 
+#include <locale>
+
 namespace platform
 {
+Locale StdLocale2Locale(std::locale loc)
+{
+  return {"",
+          "",
+          std::use_facet<std::moneypunct<char, true>>(loc).curr_symbol(),
+          std::string(1, std::use_facet<std::numpunct<char>>(loc).decimal_point()),
+          std::string(1, std::use_facet<std::numpunct<char>>(loc).thousands_sep())};
+}
+
 Locale GetCurrentLocale()
 {
-  return {"", "", ""};
+  // Environment's default locale.
+  std::locale loc;
+
+  return StdLocale2Locale(loc);
 }
+
+bool GetLocale(std::string localeName, Locale& result)
+{
+  try
+  {
+    // Set global locale
+    std::locale::global(std::locale(localeName));
```

Comment:
Why setting a global locale required? Does `std::locale(localeName)` work directly?

Reply:
You're right! There's no need to use `std::locale::global()`. Will change accordingly...

Reply:
Commit amended!!